### PR TITLE
Separate cache path

### PR DESCRIPTION
**Pull request by dhardiman**
This allows us to specify the actual cache to download the tiles to. That will allow us the opportunity to specify individual packages for each offline download we create, giving us control over when we can delete them.

### CODE REVIEW

File: MapView/Map/RMShape.h [hunk 1]
```diff
@@ -78,7 +78,7 @@
 
 @property (nonatomic, assign) BOOL scaleLineWidth;
 @property (nonatomic, assign) CGFloat shadowBlur;
-@property (nonatomic, assign) CGSize shadowOffset;
+@property (assign) CGSize shadowOffset;
```
Comment (dhardiman) [inline MapView/Map/RMShape.h:81]: This is just to fix an Xcode 6.3 warning

Comment (steveanthony) [inline MapView/Map/RMShape.h:81]: Interesting. What was the warning?

**Comment by steveanthony** [comment]
Looks good to me.

File: MapView/Map/RMShape.h (hunk 1 above)
Comment (dhardiman) [inline MapView/Map/RMShape.h:81]: It's shadowing an atomic property on `CALayer`. Surprised nobody else saw it and got irritated by it.

File: MapView/Map/RMTileCache.m [hunk 2]
```diff
@@ -423,6 +428,16 @@ - (void)beginBackgroundCacheForTileSource:(id <RMTileSource>)tileSource southWes
     }
 }
 
+- (id<RMTileCache>)databaseCacheAtPath:(NSString *)cachePath
+{
+    RMDatabaseCache *dbCache = [[RMDatabaseCache alloc] initWithDatabase:cachePath];
+    [dbCache setCapacity:LONG_MAX];
```
Comment (drhaynes) [inline MapView/Map/RMTileCache.m:434]: Presumably we have in mind a strategy for managing the tiles? As this setup will essentially eat disk space until it runs out without ever purging.

**Comment by drhaynes** [comment]
LGTM

File: MapView/Map/RMTileCache.m (hunk 2 above)
Comment (dhardiman) [inline MapView/Map/RMTileCache.m:434]: For the record, the purpose of this cache is to provide a single source package, basically a hacky way to create an MBTiles package for mapbox sources.

**Pull request merged**